js/reference.js

```diff
@@ -4132,8 +4219,95 @@ var ERMrest = (function(module) {
         }
     });
 
+    /**
+     * Constructs an Aggregate Funciton object
+     *
+     * Reference Aggregate Functions is a collection of available aggregates for the
+     * particular Reference (count for the table). Each aggregate should return an
+     * object that includes it's `alias` if it was provided and the string
+     * representation for querying the information.
+     *
+     * Usage:
+     *  Clients _do not_ directly access this constructor. ERMrest.Reference will
+     *  access this constructor for purposes of fetching aggregate data about the table.
+     * @memberof ERMrest
+     * @class
+     */
+    function ReferenceAggregateFn () {}
+
+    ReferenceAggregateFn.prototype = {
+        countAgg: function(alias) {
+            return {
+                "alias": typeof alias !== "string" ? null : alias,
+                "value": "cnt(*)"
+            };
+        }
+    };
+
+    /**
+     * Constructs an Aggregate Function object
+     *
+     * Column Aggregate Functions is a collection of available aggregates for the
+     * particular ReferenceColumn (min, max, count not null, and count distinct for it's column).
+     * Each aggregate should return an object that includes it's `alias` if it was
+     * provided and the string representation for querying for that information.
+     *
+     * Usage:
+     *  Clients _do not_ directly access this constructor. ERMrest.ReferenceColumn
+     *  will access this constructor for purposes of fetching aggregate data
+     *  for a specific column
+     * @memberof ERMrest
+     * @class
+     * @param {ERMrest.ReferenceColumn} column - the column that is used for creating column aggregates
+     */
+    function ColumnAggregateFn (column) {
+        this.column = column;
+    }
+
+    ColumnAggregateFn.prototype = {
```

Comment:
These `ColumnAggregateFn`s should either complain when the `column.isPseudo` is correct, or do something extra for those. Because `this.column.name` won't return a name that is understandable by ermrest.